lib/rules/no-restricted-imports.js

```diff
@@ -678,6 +779,35 @@ module.exports = {
 				}
 			},
 			ExportAllDeclaration: checkNode,
+			// Add support for TypeScript import equals declarations
+			TSImportEqualsDeclaration(node) {
+				if (node.moduleReference.type === "TSExternalModuleReference") {
+					const importSource = node.moduleReference.expression.value;
+
+					// Create a proper synthesized import declaration
+					const synthesizedImport = {
```

Comment:
[Refactor] 🤔 Synthesizing nodes is a rather unusual practice. It's brittle to changes in AST and AST handling, as well as verbose. It was done in the extension rule because the core rule wasn't extensible.

For this PR I think it'd be cleaner to have `checkNode` handle these TS nodes natively. There's not much logic to change:

* `checkRestrictedPathAndReport` will need to also check this node type along with `ImportDeclaration`

...and that's about it that I can see? No unit tests fail locally when I make that change and pass `node` to `checkRestrictedPathAndReport` instead of `synthesizedImport`. The `node.moduleReference.type === "TSExternalModuleReference"` check is also able to be removed.

Reply:
Makes sense, done.